### PR TITLE
[#34] [iOS] [Integrate] As a user, I can see a list of surveys on the home screen

### LINKED ISSUE #34: [iOS] [Integrate] As a user, I can see a list of surveys on the home screen
## Why

Once logged in, users can navigate through a list of surveys that they can take. The mobile app will show the basic info of the survey as they scroll through each survey.

## Acceptance Criteria

- Use the survey cover image as the background image
    - Append l to the image URL to get the high-resolution image
- For the date label, show the survey's created_at.
    - Use the date format as shown in the design on the first line
    - Use the OS relative date format on the second line
- Use the survey title as the title label
- Use the survey description as the description label
- The number of page indicators must match the number of surveys

## Design

Image

[Home](https://www.figma.com/embed?embed_host=notion&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FGjRPOjDyZ6f4EDL3wKarRK%2FChallenge-Mobile-App%3Fpage-id%3D0%253A1%26node-id%3D30%253A46%26scaling%3Dmin-zoom)

### PR DESCRIPTION
**Pull request by suho**
- Resolve #34

## What happened 👀

I used `get surveys` use case in the home screen of the iOS application

## Insight 📝

Beside the integration, I added a new serializer (FullSizeUrlSerializer) to support larger background image

## Proof Of Work 📹

https://user-images.githubusercontent.com/19943832/203217467-c8b87006-b6c2-4f70-b91c-9c4b411ecc92.mp4

### CODE REVIEW

File: ios/Surveys/Sources/Supports/Extensions/Foundation/Optional+Unwrap.swift [hunk 1]
```diff
@@ -8,3 +8,12 @@ extension Optional where Wrapped == String {
 
     var string: String { self ?? "" }
 }
+
+extension Optional {
+
+    func `let`(_ unwrapped: (Wrapped) -> Void) {
```
Comment (vnntsu) [inline ios/Surveys/Sources/Supports/Extensions/Foundation/Optional+Unwrap.swift:14]: 😘

File: ios/Surveys/Sources/Domain/KotlinSwift/UseCases/GetSurveysUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//  GetSurveysUseCase.swift
+//  Surveys
+//
+//  Created by Su Ho on 21/11/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Combine
+import Factory
+import KMPNativeCoroutinesCombine
```
Comment (vnntsu) [inline ios/Surveys/Sources/Domain/KotlinSwift/UseCases/GetSurveysUseCase.swift:11]: just concern, could we remove these imports?

**Review by vnntsu — APPROVED** [review]
LGTM! 🎉

File: ios/Surveys/Sources/Domain/KotlinSwift/UseCases/GetSurveysUseCase.swift (hunk 2 above)
Comment (suho) [reply ios/Surveys/Sources/Domain/KotlinSwift/UseCases/GetSurveysUseCase.swift:11]: Unfortunately, we can't. 🙈
Combine: AnyPublisher
Factory: @Injected
KMP: createPublisher
Shared: Shared.GetSurveysUseCase

**Review by nkhanh44 — APPROVED** [review]
LGTM ❤️

**Pull request merged**